### PR TITLE
[SPARK-35776][SQL][TESTS] Check all year-month interval types in arrow

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Add tests to check that all year-month interval types are supported in (de-)serialization from/to Arrow format.

### Why are the changes needed?
New tests should improve test coverage.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
added ut

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
@MaxGekk  Since ArrowType.Interval only support YEAR_TO_MONTH and DAY_TO_SECOND so all passed in will read out as YearMonthIntervalType

File: sql/core/src/test/scala/org/apache/spark/sql/execution/arrow/ArrowWriterSuite.scala [hunk 1]
```diff
@@ -30,7 +30,11 @@ class ArrowWriterSuite extends SparkFunSuite {
 
   test("simple") {
     def check(dt: DataType, data: Seq[Any], timeZoneId: String = null): Unit = {
-      val schema = new StructType().add("value", dt, nullable = true)
+      val avroDatatype = dt match {
+        case _: YearMonthIntervalType => YearMonthIntervalType()
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/arrow/ArrowWriterSuite.scala:34]: Sorry, I didn't get this. Do you always test only `YearMonthIntervalType()`?

Comment (AngersZhuuuu) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/arrow/ArrowWriterSuite.scala:34]: > Sorry, I didn't get this. Do you always test only `YearMonthIntervalType()`?

We can write different field of YearMonthIntervalType tp arrow, but in arrow's type it only support year_to_month
image
And the IntervalUnit only support year_to_month, 
image

So when we read data from Arrow, it only can be default YearMonthIntervalType()

**Comment by AngersZhuuuu** [comment]
retest this please

File: sql/core/src/test/scala/org/apache/spark/sql/execution/arrow/ArrowWriterSuite.scala (hunk 1 above)
Comment (MaxGekk) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/arrow/ArrowWriterSuite.scala:34]: ok. I see. Thanks.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @AngersZhuuuu .

**Pull request closed without merging**